### DISCUSSION TITLE
How add fail2ban filter for Mpd5 (https://mpd.sourceforge.net)

### DISCUSSION THREAD

**Discussion by ysha0** [fail2ban/fail2ban#3493] [General]
Hi,

How to make filter to use fail2ban with mpd5 (https://mpd.sourceforge.net), log like as:
```
Apr 14 14:10:12 srv mpd[64256]: PPTP: Incoming control connection from 112.112.112.112 51694 to 111.111.111.111 1723
--- some line of log on ---
...
--- some line of log off ---
Apr 14 14:10:14 srv mpd[64256]: [L-3] LCP: authorization failed
Apr 14 14:10:14 srv mpd[64256]: [L-3] LCP: parameter negotiation failed
```

The date and ip address clients (112.112.112.112) are on the same line, but the result of an unsuccessful authorization is a few lines later.

How to build an expression "failregex" if the arguments are on several lines of log?

**Comment by sebres** [comment]
If your fail2ban version is at least 0.10 (better 0.11.2 or higher, or 1.0.x), you can use new filter facilities that make able to match the failures across several lines by single line parsing (required a common ID for all messages of certain session).
So for the aforementioned log excerpt, one could implement the filter like this:
```ini
[Definition]

prefregex = ^\s*\S+ mpd\[<F-MLFID>\d+</F-MLFID>\]: <F-CONTENT>.+</F-CONTENT>$

failregex = ^PPTP: <F-NOFAIL>Incoming</F-NOFAIL> control connection from <ADDR>
            ^\[[^\]]+\] LCP: authorization failed
```
where `<F-NOFAIL>...</F-NOFAIL>` signaling a no failure for non-empty match, used as helper to remember IP by `<F-MLFID>...</F-MLFID>` (like numeric session id from `mpd[64256]`).

If you have fail2ban before 0.10, you can try to use multi-line parsing with `maxlines` and `<SKIPLINES>`, but note the restrictions of that, for instance https://github.com/fail2ban/fail2ban/issues/2754#issuecomment-1017529209

**Comment by ysha0** [comment]
I checked this filter on the production server and everything works Ok, thank you.

Could you include the new mpd5 filter in future versions of fail2ban?

**Comment by ysha0** [comment]
I'm create new filter /usr/local/etc/fail2ban/filter.d/mpd5.conf

```
[Definition]

prefregex = ^\s*\S+ mpd\[<F-MLFID>\d+</F-MLFID>\]: <F-CONTENT>.+</F-CONTENT>$

failregex = ^PPTP: <F-NOFAIL>Incoming</F-NOFAIL> control connection from <ADDR>
            ^\[[^\]]+\] LCP: authorization failed
```
Should I use the mpd5.conf file name or mpd5.local for it?

**Reply by sebres** [reply]
It doesn't matter... the only difference - local-file wouldn't probably conflict with a conf-file, once it'd get released and you'd obtain a copy of mpd5.conf by some upgrade later.
